pkg/build/util/util.go

```diff
@@ -436,3 +437,26 @@ func UpdateBuildEnv(build *buildapi.Build, env []kapi.EnvVar) {
 	newEnv = append(newEnv, env...)
 	SetBuildEnv(build, newEnv)
 }
+
+// FindDockerSecret looks through a set of k8s Secrets to find one that represents Docker credentials
+// and which contains credentials that are associated with the registry identified by the image.
+func FindDockerSecret(secrets []kapi.Secret, image string) *kapi.LocalObjectReference {
```

Comment:
sort-of feels like `func FindDockerSecret(secrets []kapi.Secret, image string) *kapi.Secret` would be a cleaner API